app/controllers/follows_controller.rb

```diff
@@ -68,7 +68,11 @@ def follow_params
   end
 
   def follow(followable, need_notification: false)
-    user_follow = current_user.follow(followable)
+    begin
+      user_follow = current_user.follow(followable)
+    rescue ActiveRecord::RecordInvalid => e
+      Rails.logger.error(e)
```

Comment:
We may want to do more than just log an error here, e.g. update a DataDog metric. @mstruve can point you in the right direction.

Reply:
I'm torn. I don't really think it is that important but at the same time, I have learned when in doubt log it so let's do `DataDogStatsClient.increment("users.invalid_follow")` instead of Rails.logger here to track it on the off chance we want the metric in the future for some reason.

Reply:
Tbh, I wasn't sure either, hence the "we may want to do more". But like you I lean in the direction of logging more.